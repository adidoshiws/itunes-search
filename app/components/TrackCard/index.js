/**
 *
 * Track Card
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card } from 'antd';
import T from '@components/T';
import If from '@components/If';
import { isEmpty } from 'lodash';

const CustomCard = styled(Card)`
  && {
    margin: 20px 0;
  }
`;

export function TrackCard({ trackName, collectionName, previewUrl }) {
  return (
    <CustomCard data-testid="track-card">
      <If condition={!isEmpty(trackName)} otherwise={<T data-testid="name-unavailable" id="track_name_unavailable" />}>
        <T data-testid="name" id="track_name" values={{ trackName: trackName }} />
      </If>
      <If
        condition={!isEmpty(collectionName)}
        otherwise={<T data-testid="collectionName-unavailable" id="track_collection_name_unavailable" />}
      >
        <T data-testid="collectionName" id="track_collection_name" values={{ collectionName: collectionName }} />
      </If>
      <If condition={previewUrl} otherwise={<T data-testid="preview-unavailable" id="track_preview_unavailable" />}>
        <T data-testid="preview" id="track_preview" values={{ preview: previewUrl }} />
      </If>
    </CustomCard>
  );
}

TrackCard.propTypes = {
  trackName: PropTypes.string,
  collectionName: PropTypes.string,
  previewUrl: PropTypes.string
};

export default TrackCard;
