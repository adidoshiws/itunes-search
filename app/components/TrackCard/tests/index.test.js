/**
 *
 * Tests for TrackCard
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import TrackCard from '../index';
import { translate } from '@app/components/IntlGlobalProvider/index';

describe('<TrackCard />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<TrackCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 TrackCard component', () => {
    const { getAllByTestId } = renderWithIntl(<TrackCard />);
    expect(getAllByTestId('track-card').length).toBe(1);
  });

  it('should render the track details inside the card', () => {
    const trackName = 'Upside Down';
    const collectionName = 'Jack Johnson and Friends';
    const previewUrl =
      'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/5e/5b/3d/5e5b3df4-deb5-da78-5d64-fe51d8404d5c/mzaf_13341178261601361485.plus.aac.p.m4a';
    const { getByTestId } = renderWithIntl(
      <TrackCard trackName={trackName} collectionName={collectionName} preview={previewUrl} />
    );
    expect(getByTestId('trackName')).toHaveTextContent(trackName);
    expect(getByTestId('collectionName')).toHaveTextContent(collectionName);
    expect(getByTestId('preview')).toHaveTextContent(previewUrl);
  });

  it('should render the track unavailable messages in case any props are unavailable or have falsy values', () => {
    const trackUnavailable = translate('track_name_unavailable');
    const collectionNameUnavailable = translate('track_collection_name_unavailable');
    const previewUnavailable = translate('track_preview_unavailable');
    const { getByTestId } = renderWithIntl(<TrackCard />);
    expect(getByTestId('name-unavailable')).toHaveTextContent(trackUnavailable);
    expect(getByTestId('collectionName-unavailable')).toHaveTextContent(collectionNameUnavailable);
    expect(getByTestId('preview-unavailable')).toHaveTextContent(previewUnavailable);
  });
});
