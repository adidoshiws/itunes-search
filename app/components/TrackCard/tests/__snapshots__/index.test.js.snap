// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-col ant-col-6 TrackCard__CustomCard-sc-10bk196-0 ilIdfW"
      data-testid="track-card"
    >
      <a
        href="/tracks/undefined"
        style="color: black;"
      >
        <p
          class="T__StyledText-gjlic1-0 dmkfex"
          data-testid="img-unavailable"
        >
          Track img is unavailable
        </p>
        <p
          class="T__StyledText-gjlic1-0 dmkfex"
          data-testid="name-unavailable"
        >
          Track name is unavailable
        </p>
        <p
          class="T__StyledText-gjlic1-0 dmkfex"
          data-testid="collectionName-unavailable"
        >
          Track collection name unavaiable
        </p>
      </a>
      <p
        class="T__StyledText-gjlic1-0 dmkfex"
        data-testid="preview-unavailable"
      >
        Track preview is unavaiable
      </p>
    </div>
  </div>
</body>
`;
